qiskit_machine_learning/kernels/parameterized_quantum_kernel.py

```diff
@@ -0,0 +1,314 @@
+                           y_vec=None):
+        r"""
+        docstring
+        """
+
+        assert x_vec is not None
+        assert y_vec is not None
+        from qiskit_machine_learning.algorithms import QSVC    
+        
+        # check that params are the right dimension
+        self.bind_free_params(params)
+        
+        qsvc = QSVC(quantum_kernel = self)
+        qsvc.fit(x_vec, y_vec)
+        score = qsvc.score(x_vec, y_vec)
+
+        # The dual coefficients are equal
+        # to the Lagrange multipliers, termwise
+        # multiplied by the corresponding datapoint's 
+        # label. Only nonzero coefficients are returned
+        a = qsvc.dual_coef_[0]
+        
+        # Get the indices of our data which correspond to
+        # support vectors
+        sv = qsvc.support_
+        
+        #Filter out entries for non-support vectors
+        K = self.evaluate(x_vec)[sv,:][:,sv]
+
+        # The -1 is here because qiskit 
+        # optimizers minimize by default
+        return np.sum(np.abs(a)) - 1/2 * (a.T @ K @ a) + np.sum(a)
+
+    def _model_complexity(self, params, x_vec=None, y_vec=None):
+        r"""
+        docstring
+        """
+
+        assert x_vec is not None
+        assert y_vec is not None
+        
+        # check that params are the right dimension
+        self.bind_free_params(params)
+        K = self.evaluate(x_vec)
+        y = np.array(y_vec)
+        
+        # The -1 is here because qiskit 
+        # optimizers minimize by default
+        return -1 * y.T @ np.linalg.pinv(K) @ y
+
+    def train_kernel(self,
+                     objective_function = 'alignment',
+                     optimizer = SPSA(),
+                     x_vec=None,
+                     y_vec=None,
+                     bounds = None,
+                     initial_point = None,
+                     gradient_fn = None):
+
+        # objective_function(params) --> scalar
+
+        if isinstance(objective_function, str):
+            assert x_vec is not None and y_vec is not None, "x_vec and y_vec must be supplied " \
+                   "if a pre-made objective function is chosen."
+            if objective_function == 'weighted_alignment':
+                obj_fun = partial(self._weighted_alignment, x_vec=x_vec, y_vec=y_vec)
+            elif objective_function == 'alignment':
+                obj_fun = partial(self._alignment, x_vec=x_vec, y_vec=y_vec)
+            elif objective_function == 'model_complexity':
+                obj_fun = partial(self._model_complexity, x_vec=x_vec, y_vec=y_vec)
+            else:
+                raise ValueError('{} is not a valid choice of objecive function'\
+                                 ''.format(objective_function))
+        elif isinstance(objective_function, Callable):
+            obj_fun = objective_function
+
+        nparams = len(self._free_params)
+        if initial_point is None:
+            initial_point = np.random.rand(nparams)
```

Comment:
Instead of directly using numpy global random it should be using algorithm_globals, which is a global Generator in Qiskit that we can seed for reproducibility and all algorithms etc use eg here https://github.com/Qiskit/qiskit-machine-learning/blob/8a6e73beca8175e782d17d562f8e7b8642f7697a/qiskit_machine_learning/algorithms/trainable_model.py#L206